docs/Fleet management Design.md

```diff
@@ -101,52 +125,122 @@ type ResourceList map[ResourceName]resource.Quantity
 
 ```
 
-### Design details
+## Design details
 Here is how we handle the following functionalities in the OCM in our new controller. The overall idea is that 
 a `managedCluster` CR represents a member cluster. A cluster joins the fleet when a corresponding `managedCluster` CR is
 successfully applied to the hub cluster. Similarly, a member cluster leaves the hub cluster when its corresponding 
 `managedCluster` CR is removed from the cluster. 
 
-#### Join a member cluster to the fleet
+### Join a member cluster to the fleet
 The managedCluster will look for a secret named `name of the cluster`-`kubeconfig`. 
 For example, the controller that reconciles a `managedCluster` called "member-cluster-A" will look for
-a secret named `member-cluster-A-kubeconfig`. The secret contains the entire kubeconfig data. The controller will then 
-create a namespace of the same name if it does not already exist.
+a secret named `member-cluster-A-kubeconfig`. Initially, the secret can simply contain the entire kubeconfig data.
+
+The controller keeps an in-memory map that keys on the cluster name to track all the information belong to the cluster. 
+One of the information can be the kubeclient created from the secret. Here are a list of operations a controller will 
+perform in the member cluster joining process
+
+1. The controller "claims" the member cluster by creating a "lease" configMap in the member cluster.
+   * The configMap's name is a well known name by convention.
+   * It contains the hub cluster's [unique ID](https://github.com/kubernetes/kubernetes/issues/77487) which is the UID of the namespace kube-system for now.
+   * The controller will reject the member cluster if the configMap already contains a cluster ID not the same as its ID.
+   * We could use a CR but that means we also need to apply a CRD there.
+2. The controller creates a namespace of the same name as the member cluster in the hub cluster.
+3. The controller starts a health check loop for each member cluster which
+   * Checks the health of the member cluster by updating the last health check time in the lease configMap
+   * Lists all the nodes in a member cluster and aggregates their `Capacity` and `Allocatable`
+   * Updating the `managedCluster` condition status
+
+### API definition change proposal
```

Comment:
Define the lease with standard param names in this section as it is also a critical part of the API?